tests/utils.py

```diff
@@ -33,13 +34,13 @@ def out_kwarg_test(f, args, kwargs={}, **assert_kwargs):
 
   def torch_function(*torch_args):
     out1 = f(*torch_args, **torch_kwargs)
-    out2 = torch.zeros_like(out1)
+    out2 = torch_tree_map(torch.zeros_like, out1)
     out3 = f(*torch_args, out=out2, **torch_kwargs)
     return out1, out2, out3
 
   jax_function = t2j(torch_function)
   jax_out1, jax_out2, jax_out3 = jax_function(*args)
-  assert jax_out2 is jax_out3
+  assert all(jax.tree.leaves(jax.tree.map(lambda x, y: x is y, jax_out2, jax_out3)))
```

Comment:
is it actually necessary to change this? i would expect `assert jax_out2 is jax_out3` to work as well

Reply:
Because when converting to Jax the out2 and out3 gets converted separately, but the array stored inside is the same copy.